src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -353,7 +354,19 @@ theorem msb_eq_getLsbD_last (x : BitVec w) :
 @[bv_toNat] theorem getLsbD_succ_last (x : BitVec (w + 1)) :
     x.getLsbD w = decide (2 ^ w ≤ x.toNat) := getLsbD_last x
 
-@[bv_toNat] theorem msb_eq_decide (x : BitVec w) : BitVec.msb x = decide (2 ^ (w-1) ≤ x.toNat) := by
+
+/-- 
+An alternative to `msb_eq_decide` in terms of `2 * x.toNat`,
+in order to avoid `2 ^ (w - 1)`.
+-/
+@[bv_toNat] theorem msb_eq_decide_le_mul_two (x : BitVec w) : 
+    BitVec.msb x = decide (2 ^ w ≤ 2 * x.toNat) := by
+  rw [x.msb_eq_getLsbD_last, x.getLsbD_last]
+  simp
+  rcases w with rfl | w  <;> simp <;> omega
+
+@[bv_toNat, deprecated msb_eq_decide_le_mul_two (since := "21-10-2024") ]
+theorem msb_eq_decide (x : BitVec w) : BitVec.msb x = decide (2 ^ (w-1) ≤ x.toNat) := by
```

Comment:
Do we want to give it a new name? The old theorem statement was against conventions. I feel we can fix it inline.

Reply:
I decided to change `msb_eq_decide` to be the more natural definition.